File: spring-data/src/main/java/com/mongodb/ms0/example/springdata/models/Customer.java
package com.mongodb.ms0.example.springdata.models;



import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

/*
We are explictly defining a value for the document annotation here because the collection is named differently than Spring-data's default. Spring assumes class name and collection name are the same. So the use of the plular for Customer requires this correction
 */
@Document("customers")
public class Customer {

    @Id
    private String id;
    private String firstName;
    private String lastName;
    private String title;
    private Address address;

    private List<Phone> phones;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }

    public List<Phone> getPhones() {
        return phones;
    }

    public void setPhones(List<Phone> phones) {
        this.phones = phones;
    }
}

File: dotnet-sample/Models/Customer.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace dotnet_sample;

[BsonIgnoreExtraElements]
public class Customer
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? id { get; set; }
    public string lastName { get; set; }
    public string firstName { get; set; }
    public string title { get; set; }
    public Address address { get; set; }
    public List<Phone> phones { get; set; }
    
    
}